File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-2.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }

head -> 1 -> 2 -> 3


 */
func removeNthFromEnd(head *ListNode, n int) *ListNode {
	if head == nil{
		return nil
	}
	var prev *ListNode
	counter := head
	length := 0
	for counter != nil{
		length += 1
		counter = counter.Next
	}
	fmt.Println(length)
	lengthFromStart := length - n
	cur := head

	for i := 1; i<= lengthFromStart; i++{
		temp := cur
		cur = cur.Next
		prev = temp
	}
	fmt.Println(prev, cur)

	if prev == nil{
		// remove first element
		head = head.Next
	}else{
		prev.Next = cur.Next
	}



	return head
}

File: Data Structures & Algorithms/linked-list-cycle-detection/submission-2.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        s = head
        f = head
        while f != None:
            s = s.next
            if f.next == None:
                return False
            f = f.next.next
            if s == f:
                return True
        return False
        